### PR TITLE
ES6fy and simplify code

### PR DESCRIPTION
**Pull request by xgrommx**

### CODE REVIEW

File: src/components/Json-results.js [hunk 1]
```diff
@@ -1,38 +1,35 @@
 import React from 'react';
 import Intent from '../intent/json-intent';
 
-var JsonResults = React.createClass({
+export default class extends React.Component {
```
Comment (Cmdv) [inline src/components/Json-results.js:4]: I'll be keeping away from the Class in ES6

https://medium.com/javascript-scene/how-to-fix-the-es6-class-keyword-2d42bb3f4caf

File: src/index.js [hunk 2]
```diff
@@ -12,51 +12,54 @@ var DefaultRoute = Router.DefaultRoute,
...
 
-var App = React.createClass({
-  render: function () {
-    return (
-      <div className="container theme-showcase">
-        <nav className="navbar navbar-default">
-          <div className="container">
-            <div className="navbar-header">
-              <Link className="navbar-brand" to="app">React RxJS</Link>
+class App extends React.Component {
```
Comment (Cmdv) [inline src/index.js]: Same reasoning about ES6 class

File: src/routes/About.js [hunk 3]
```diff
@@ -1,28 +1,25 @@
 import React from 'react';
 import Intent from '../intent/Counter-intent';
 
-var About = React.createClass({
-
+export default class extends React.Component {
```
Comment (Cmdv) [inline src/routes/About.js:4]: ES6 Class

File: src/routes/Home.js [hunk 4]
```diff
@@ -1,31 +1,28 @@
 import React from 'react';
 import Intent from '../intent/Counter-intent';
 
-var Home = React.createClass({
-
+export default class extends React.Component {
```
Comment (Cmdv) [inline src/routes/Home.js:4]: ES6 Class

File: src/routes/Json.js [hunk 5]
```diff
@@ -2,30 +2,24 @@ import React from 'react';
 import Intent from '../intent/json-intent';
 import Results from '../components/Json-results';
 
-var JsonResults= React.createClass({
-
+export default class extends React.Component {
```
Comment (Cmdv) [inline src/routes/Json.js:5]: ES6 Class

File: src/routes/Login.js [hunk 6]
```diff
@@ -1,14 +1,11 @@
 import React from 'react';
 
-var Login = React.createClass({
-
+export default class extends React.Component {
```
Comment (Cmdv) [inline src/routes/Login.js:3]: ES6 Class

**Comment by Cmdv** [comment]
Really liking the implementations does definitely make parts clearer, I was going to also go over the code and make sure that Fat arrows `=>` wasn't used everywhere because the use of them makes the function anonymous and sometimes you need to be able to view the function name in the Stack trace. 

I'll have a deeper look but pretty sure your implementation the functions were already anonymous.

Do you agree with not using the Class in ES6 as mention in code comment?

But thank you for the input will really make the code a lot cleaner :smile:

**Comment by BerkeleyTrue** [comment]
While using es6 classes vs createClass is more opinion then anything else the only real reason to use createClass is if you want to support mixins. But there is numerous [discussion](https://github.com/facebook/react/issues/4023) on whether mixins are bad(they are IMHO).

I personally love using es6 classes because the code looks cleaner.

**Pull request merged**